### PR TITLE
Checking ipywidgets is installed for ensure tqdm working

### LINKED ISSUE #1687: name 'IProgress' is not defined
name 'IProgress' is not defined when running from Jupyter notebook.

### PR DESCRIPTION
**Pull request by olineumann**
## What does this PR do?

Some users having problems with the progress bar in combination with jupyter notebook/lab. Sometimes it crashes because of missing dependencies, sometimes the progress bar isn't displayed correctly. Normally tqdm should select the correct package via tqdm.auto which is used in the code, but as some users reporting (not only in pytorch_lightning) it seems not working correctly.

Since pytorch_lightning doesn't want to add ipywidgets (which often is the problem) to the dependencies it would be better to check if this package is installed before importing tqdm.auto and import only tqdm of not.

As an example run this in jupyter without ipywidgets installed:
```python
import time

# this will fail
from tqdm.auto import tqdm

# this not
#try:
#    import ipywidgets
#    from tqdm.auto import tqdm
#except ImportError as e:
#    from tqdm import tqdm

for i in tqdm(range(1000)):
    time.sleep(0.01)
```

Fixes #1687 

# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [x] Did you make sure to update the documentation with your changes?
- [x] Did you write any new necessary tests? 
Not needed I think? 
- [x] Did you verify new and existing tests pass locally with your changes?
Tests passed locally.
- [x] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?
Not necessary.

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by olineumann** [comment]
Just searched for `tqdm.auto` imports and replaced it. I think in progress.py it is correct in lr_finder.py I'm not sure and I'm wondering why lr_finder doesn't use ProgressBar?

File: pytorch_lightning/callbacks/progress.py [hunk 1]
```diff
@@ -7,7 +7,14 @@
 """
 import sys
 
-from tqdm.auto import tqdm
+try:
+    # try importing ipywidgets before importing tqdm
+    # to ensure it won't fail and a progress bar is displayed
+    import ipywidgets
+    from tqdm.auto import tqdm
+except ImportError as e:
```
Comment (awaelchli) [inline pytorch_lightning/callbacks/progress.py:15]: could we use 
```python
import importlib 
if importlib.util.find_spec("ipywidgets") is not None:
    from tqdm.auto import tqdm
else:
    from tqdm import tqdm
```
? (brain compiled, may have typos)
ImportError is not suitable here in general.

Comment (awaelchli) [reply pytorch_lightning/callbacks/progress.py:15]: should probably also add "pragma no cover " not to decrease coverage

Comment (olineumann) [reply pytorch_lightning/callbacks/progress.py]: Yes, sure. Thanks!

**Comment by williamFalcon** [comment]
@olineumann love the PR! this is a great addition

**Pull request merged**